src/svm/transactionUtils.ts

```diff
@@ -51,13 +52,14 @@ export async function sendTransactionWithLookupTable(
       addresses: lookupAddresses.slice(i, i + maxExtendedAccounts),
     });
 
-    await web3.sendAndConfirmTransaction(connection, new web3.Transaction().add(extendInstruction), [sender], {
-      skipPreflight: true, // Avoids recent slot mismatch in simulation.
-    });
+    await web3.sendAndConfirmTransaction(connection, new web3.Transaction().add(extendInstruction), [sender]);
   }
 
-  // Avoids invalid ALT index as ALT might not be active yet on the following tx.
-  await new Promise((resolve) => setTimeout(resolve, 1000));
+  // Wait for slot to advance. LUTs only active after slot advance.
+  const initialSlot = await connection.getSlot();
```

Comment:
Nice finding!
Suggestion: Could we use something like this [connection.onSlotChange](https://solana-labs.github.io/solana-web3.js/classes/Connection.html#onslotchange) to subscribe to slot changes? Not sure how reliable onSlotChange is though

Reply:
I tried this and this was better. not sure if onSlotChange was delayed or returned early but it did to result in the tests always working as expected, like this implementation does.